vignettes/pull_request_comments.Rmd

```diff
@@ -0,0 +1,249 @@
+rates at [its
+website](https://docs.github.com/en/free-pro-team@latest/rest/overview/resources-in-the-rest-api#rate-limiting).
+
+If using a personal account token from a free GitHub account, the number
+of API calls per hour increases to 5000 **per hour**. Therefore, it is
+recommended you create a personal token by following the [GitHub
+Documentation
+instructions](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token#:~:text=Creating%20a%20token.%201%20Verify%20your%20email%20address%2C,able%20to%20see%20the%20token%20again.%20More%20items).
+The process should not take more than 2 minutes.
+
+The token should then be copied and pasted into `../.ssh/github_token`.
+Both the `.ssh` folder and `github_token` file may need to be created
+manually. Check for hidden folders before creating the `.ssh` folder.
+The `github_token` file should not have any file extensions.
+
+## Libraries
+
+To use the notebook, you will need to ensure you have the required R
+packages installed.
+
+```{r warning=FALSE,message=FALSE}
+rm(list = ls())
+require(kaiaulu)
+require(data.table)
+require(jsonlite)
+require(knitr)
+require(magrittr)
+require(gt)
+```
+
+## Project Configuration File
+
+To use the pipeline, you must specify the organization and project of
+interest, and your token.
+
+```{r warning=FALSE}
+conf <- parse_config("../conf/kaiaulu.yml")
+owner <- get_github_owner(conf, "project_key_1") # Has to match github organization (e.g. github.com/sailuh)
+repo <- get_github_repo(conf, "project_key_1") # Has to match github repository (e.g. github.com/sailuh/perceive)
+
+# Path you wish to save all raw data.
+save_path_issue_refresh <- get_github_issue_search_path(conf, "project_key_1")
+save_path_issue <- get_github_issue_path(conf, "project_key_1")
+save_path_pull_request <- get_github_pull_request_path(conf, "project_key_1")
+save_path_pr_comments <- get_github_pr_comments_path(conf, "project_key_1")
+save_path_issue_or_pr_comments <- get_github_issue_or_pr_comment_path(conf, "project_key_1")
+save_path_commit <- get_github_commit_path(conf, "project_key_1")
+save_path_pr_reviews <- get_github_pr_review_path(conf, "project_key_1")
+save_path_pr_commits <- get_github_pr_commits_path(conf, "project_key_1")
+save_path_pr_files <- get_github_pr_files_path(conf, "project_key_1")
+save_path_pr_reviewers <- get_github_pr_reviewers_path(conf, "project_key_1")
+save_path_pr_merge <- get_github_pr_merge_path(conf, "project_key_1")
+# Create all folder directories
+#create_file_directory(conf)
+
+# your file github_token (a text file) contains the GitHub token API
+token <- scan("~/.ssh/github_token",what="character",quiet=TRUE)
+```
+
+# GitHub Pull Request Comments
+
+## Initial Pull Request Comment
+
+The first pull request comment in the conversations tab is about the
+pull request itself. To get this, we must utilize the pull request
```

Comment:
This is confusing. All comments in the pull request are about the pull request. What you want to say here is that GitHub treats the first comment of a pull request as a pull request *description*, and will therefore require its own endpoint to be obtained instead of being available under PR comments endpoint. 

While chronologically it makes sense to discuss the PR first comment here first, it may be easier to explain why you are using it *after* you use issue_or_pr_comments. At that point, the comments are already downloaded, and you can quickly show the first comment is absent, prompting you to explain the PR endpoint.